restapi.go

```diff
@@ -233,26 +235,24 @@ func (self *RestAPI) pushNotification(reqId string, kv map[string]string, perdp
 		case "service":
 			// three keys need to be ignored
 		case "badge":
-			if v != "" {
-				var e error
-				_, e = strconv.Atoi(v)
-				if e == nil {
-					notif.Data["badge"] = v
-				} else {
-					notif.Data["badge"] = "0"
-				}
+			if _, err := strconv.Atoi(v); err == nil {
+				notif.Data["badge"] = v
+			} else {
+				notif.Data["badge"] = "0"
 			}
 		default:
 			notif.Data[k] = v
 		}
 	}
 
+	logger.Debugf("Parameters accepted: %+v", notif)
+
 	if notif.IsEmpty() {
-		logger.Errorf("RequestId=%v From=%v Service=%v EmptyNotification", reqId, remoteAddr, service)
+		logger.Errorf("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs, err)
 		return
 	}
 
-	logger.Infof("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\"", reqId, remoteAddr, service, len(subs), subs)
+	logger.Infof("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs)
```

Comment:
I punctuate. It's an ESL thing, I think. I think I read some go blog post
that discouraged punctuation. It could certainly make analyzing blogs more
interesting.

Tina
On Jun 9, 2015 9:16 PM, "Misha Nasledov" <notifications@github.com> wrote:

> In restapi.go
> <https://github.com/uniqush/uniqush-push/pull/66#discussion_r32087033>:
>
> >  		return
> >  	}
> >
> > -	logger.Infof("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\"", reqId, remoteAddr, service, len(subs), subs)
> > +	logger.Infof("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs)
>
> why the extra .?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uniqush/uniqush-push/pull/66/files#r32087033>.
>